src/CodeGen/GenerateSyntax.hs

```diff
@@ -13,32 +13,62 @@ module CodeGen.GenerateSyntax
 
 import Data.Char
 import Language.Haskell.TH
+import Data.HashSet (HashSet)
 import Language.Haskell.TH.Syntax as TH
 import CodeGen.Deserialize (MkDatatype (..), MkDatatypeName (..), MkField (..), MkRequired (..), MkType (..), MkNamed (..), MkMultiple (..))
 import Data.List.NonEmpty (NonEmpty (..))
 import Data.Foldable
 import Data.Text (Text)
 import qualified Data.HashSet as HashSet
-import Data.HashSet (HashSet)
+import qualified TreeSitter.Importing as TS
+import GHC.Generics hiding (Constructor, Datatype)
+import Foreign.Ptr
+import qualified TreeSitter.Language as TS
+import Foreign.C.String
+import Data.Proxy
 
 -- Auto-generate Haskell datatypes for sums, products and leaf types
-datatypeForConstructors :: MkDatatype -> Q Dec
-datatypeForConstructors (SumType (DatatypeName datatypeName) named subtypes) = do
+datatypeForConstructors :: Ptr TS.Language -> MkDatatype -> Q [Dec]
```

Comment:
1. I think we can probably rename this. `datatypeForConstructors` doesn’t really reflect either what it takes or what it does.

2. This would probably be the place to define an `astDeclarationsForLanguage` or similar function (see above).